practice-two/src/components/FormValidation/FormValidation.jsx

```diff
@@ -67,10 +88,22 @@ const FormValidation = (props) => {
       setErrors(newErrors);
 
       if (Object.keys(newErrors).length === 0) {
-        onShowForm();
+        const response = await createCustomerService(formData);
+
+        if (response.error) {
+          handleShowForm();
+          showToastInfo(MESSAGES.GET.ERRORS.ADD_FAILED);
+          resetForm();
+          return;
+        } else {
+          handleShowForm();
+          dispatch(createCustomer(response.data));
+          showToastInfo(MESSAGES.GET.SUCCESSES.ADD_SUCCESSED);
+          resetForm();
```

Comment:
both if-else have showToastInfo and resetForm please cache value and do it after if-else condition

Reply:
I move handleShowForm into resetForm function 
![image](https://github.com/datdragon777/React-Training/assets/74806826/129a3bb1-da62-4a59-a70e-78078c45a8a0)